src/main/java/com/mongodb/jdbc/MySQLResultSetMetaData.java

```diff
@@ -3,12 +3,29 @@
 import java.sql.ResultSetMetaData;
 import java.sql.SQLException;
 import java.sql.Types;
-import org.bson.BsonType;
+import java.util.HashMap;
+import java.util.List;
 
 public class MySQLResultSetMetaData extends MongoResultSetMetaData implements ResultSetMetaData {
+    protected List<Column> columns;
+    protected HashMap<String, Integer> columnPositions;
 
     public MySQLResultSetMetaData(MongoResultDoc metadataDoc) {
-        super(metadataDoc);
+        columns = metadataDoc.columns;
```

Comment:
This was just moved from the super class since it was specific to `MySQL`, as mentioned above.